library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -175,8 +181,7 @@ protected void initializePreparedStatement()
     try {
       // If its a range query pass upper and lower bounds, If its a polling query pass only the lower bound
       if (isPollerPartition) {
-        ps = store.getConnection().prepareStatement(buildRangeQuery(rangeQueryPair.getKey(), Integer.MAX_VALUE),
-            TYPE_FORWARD_ONLY, CONCUR_READ_ONLY);
+        lastOffset = rangeQueryPair.getKey();
       } else {
         ps = store.getConnection().prepareStatement(
```

Comment:
As you have moved  part to initialize prepared statement from "if" and not from "else" the code doesn't look clean. Anyway it was just a suggestion.